docs/python/tutorial-fastapi.md

```diff
@@ -0,0 +1,652 @@
+5. From the drop-down menu, select the **FastAPI** configuration option from the drop-down list:
+![Drop down with debugger configuration options, with FastAPI being highlighted.](images/fastapi-tutorial/fastapi_debug_config_option.png)
+
+    This will invoke uvicorn to start the application server through the debugger, which will allow us to step through the code to inspect its behavior. You will see something like the following in the terminal:
+
+    ![Uvicorn server running message displayed in the terminal, with an URL to access the app.](images/fastapi-tutorial/fastapi_debug_terminal.png)
+
+6. `Ctrl` + click the http://127.0.0.1:8000/ URL in the terminal to open your default browser to that address:
+![Hello World message displayed in the browser.](images/fastapi-tutorial/helloworld_browser.png)
+
+    Congrats! Your FastAPI app is up and running!
+
+7. Stop the debugger by clicking on the stop icon in the debug toolbar, or through `kb(workbench.action.debug.stop)`.
+
+## Create a model for grocery list items
+Now that we have the FastAPI app working, let’s define our grocery list items by creating a Pydantic model.
+Pydantic is a data validation and parsing library that integrates seamlessly with FastAPI. It allows you to define data models using Python classes with [type hints](https://docs.python.org/3/library/typing.html) for automatic validation and parsing of incoming data (which are called payloads) in API requests.
+
+
+Pylance, the default language server for Python in VS Code, supports type hinting features that can be helpful for working with Pydantic models and FastAPI. This is because Pylance is built on top of [Pyright](https://github.com/microsoft/pyright), which is a static type checker for Python that can detect type errors in your code to prevent bugs and improve code quality.
+
+The 3 steps below are optional, but given FastAPI leverages type hints extensively to improve code readability and validation, we can take advantage of Pylance's type checking features to catch errors early on:
+
+1. Open the Settings UI page (`kb(workbench.action.openSettings)`)
+2. Search for "python type checking mode" and set it to "basic" to enable basic type checking. This will enable Pylance to show diagnostics and warnings to catch simple type-related errors. Alternatively, you can set it to "strict" to enable more advanced [type checking rules](https://microsoft.github.io/pyright/#/configuration?id=diagnostic-rule-defaults).
+    ![Python Analysis Type Checking Mode options (off, basic and strict) in settings UI page.](images/fastapi-tutorial/type_checking_mode_setting.png)
+3. Next, search for "Python inlay type hints", and enable inlay hints for Variable Types and Function Return Types:
+    ![Two Python Analysis Type Hints settings being enabled in the settings UI page: for Function Return Types and for Variable Types.](images/fastapi-tutorial/function_and_variable_return_type_hint_settings.png)
```

Comment:
Since this is marked as optional, do you think we should put the steps for adding the model before this section? My thinking is that it allows the user to get to the crucial information quicker but I also see that enabling these settings makes sense before adding the model for the users that do this step. Or maybe this could be formatted as a 💡Tip? +0 but would love to hear your thoughts.

Reply:
Yeah that's a great point. I swapped these two instruction blocks within this section!